### ISSUE TITLE
ui.notify and ui.notification

### ISSUE THREAD

**Issue by kwmartin** [zauberzeug/nicegui#2773]
### Description

1) ui.notification: timeout is ignored, error is tripped after 1s internal timeout default which is not affected by setting timeout
2) ui.notify timeout is sometimes ignored (probably due to asyncio). Note: timeout is ms, not sec. A timeout of None is actually 0, should probably be a large number. When inside an async function, ui.notify disappears too quickly to be seen.
3) Ugly workaround, place inside async function, use a large timeout (maybe 5000), and follow immediately by await asyncio.sleep(desired_timeout - in sec float not msec - maybe 2.5).
4) close-button default color (blue) is almost impossible to see when color is 'positive' (green)

**Comment by falkoschindler** [comment]
Hi @kwmartin,

1. This works for me:
    ```py
    ui.button('Say Hi!', on_click=lambda: ui.notification('Hi!', timeout=3))
    ```

2. I don't understand:
    - What do you mean with "probably due to asyncio"? How to reproduce the problem?
    - No, we expect the timeout to be given in seconds.
    - "A timeout of None is actually 0, should probably be a large number." No, Quasar interprets 0 as "never dismiss automatically", but we think `None` is more intuitive for this case.
    - "When inside an async function, ui.notify disappears too quickly to be seen." How to reproduce this problem?

3. Please share some example code.

4. Yes, that's right, but hard to fix. It depends on the primary color of the user's app, which we don't control. And neither changing NiceGUI's default primary color nor changing Quasar's "positive" color seems like a reasonable option. As a workaround, the user can easily set a different color.

**Comment by kwmartin** [comment]
Getting a minimal example is very difficult for me; I am attempting to run both fastap routing and nicegui routing and with tls, so tearing apart to a small example which illustrates the problem would take me a couple of days, which I don't currently have. I have included a code snippet below around which the issues occurred. The snippet has a lot of commented out lines which I left in to give an idea of what I was trying. The logging lines include printing out times, but they are not all flushed at the expected times. The probably due to asyncio is just a rough guess without enough evidence to support, but I have seen many times that when the function is made async, then the notify is not seen unless an await is included. Re the None vs 0, I'm more use to the Quasar approach, but that is a "style" choice. Still, I haven't seen in the documentation that None indicates to immediately timeout? Just a suggestion. Re 4, yes one can set a different color, but looking up how to do it for the text is a pain for the user. I assume the easiest way is to add a style to something (not sure what). When customizing nicegui, one needs to be proficient in nicegui, quasar, vue, css, and html. A tall task for a newbie (which I am for front ends), and they all have sometimes very different approaches to documentation (especially Quasar vs Vue). Anyways here is my code as it now stands that seems to work. There must be something really unexpected happening to account for the ms vs s issue. Please note that the line:
```
ui.notify("Code is incorrect", color='negative', timeout=2500, close_button='Retry')
```
does timeout in 2.5 s in my code!
```
@ui.page('/two_factor')
def two_factor(request: Request) -> Optional[RedirectResponse]:
    glbls.nmb_attempts = 0
    print(f"*************** In /two_factor, nmb_attempts: {glbls.nmb_attempts}")
    async def chck_code() -> None:
        if int(code.value) == app.storage.user['two_fctr']:
            # glbls.logger.debug('Before ui.notify')
            n = ui.notify("Successfully Registered", timeout=5000, close_button='OK', color='positive')
            # glbls.logger.debug('After ui.notify')
            await asyncio.sleep(3)
            # glbls.logger.debug('After asyncio.sleep(20)')
#            n = ui.notification("Successfully Registered", timeout=60000, close_button='OK', color='positive')
#            for i in range(10):
#                n.message = f'"Successfully Registering" {i/10:.0%}'
#                n.spinner = True
#                await asyncio.sleep(0.5)
#            n.message = 'Done!'
#            n.spinner = False
#            await asyncio.sleep(5)
#            n.dismiss()
            ui.navigate.to('https://remoteker.health:8001')
        else:
            glbls.nmb_attempts += 1
            ui.notify("Code is incorrect", color='negative', timeout=2500, close_button='Retry')
            # breakpoint()
            if glbls.nmb_attempts >= 4:
                ui.navigate.to('https://remoteker.health:8001')
           
    with ui.card().classes('absolute-center'):
        code = ui.input('Code From Phone Text').on('keydown.enter', chck_code)
        ui.button('Resend Code', on_click=send_code)
```
If I can get more insight into what is happening, I'll reopen. Finally one other issue is I can't figure out why sometimes updates to request.session are ignored, sometimes request.session is saved between sessions, and why id(request) is giving different values all over the place; there seems to be many "requests" around; this may be a fastapi vs starlette vs nicegui issue; Since I haven't figured out what is handling the socket logging, I haven't been able to track it down and debugging with asyncio is a bear; irrespective, almost certainly a separate issue. Currently, I'm returning values through nicegui's app.storage.user to the fastapi side which seems to be working (I'm keeping my fingers crossed re garbage collection closure?)

**Comment by falkoschindler** [comment]
Here is an example based on your commented code, which in my point of view works as expected:
```py
async def demo():
    n = ui.notification('Registering...', timeout=None, spinner=True)
    for i in range(10):
        n.message = f'Registering {i/10:.0%}'
        await asyncio.sleep(0.5)
    n.message = 'Done!'
    n.spinner = False
    await asyncio.sleep(5)
    n.dismiss()

ui.button('Demo', on_click=demo)
```

Regarding the timeout unit:
- The `ui.notify` function doesn't really have a `timeout` parameter, but only offers to forward arbitrary keyword arguments to Quasar's `notify` function. And because Quasar expects the timeout to be set in milliseconds, this is indeed the unit for `timeout` in `ui.notify`.
- The `ui.notification` element, which we added later, allows to change properties after its creation. Here we chose to use the SI unit "seconds" for the `timeout` parameter. We think this is the more natural unit for a Python API.

We could change the unit for `ui.notify`, but this would be a nasty breaking change and we chose not to do that.

Regarding the color of the close button, I suggest you to open a new feature request or pull request with a specific proposal of what you would like to change. Thanks!
